src/rgw/rgw_rest_pubsub.cc

```diff
@@ -676,8 +676,8 @@ class RGWPSListNotifs_ObjStore_S3 : public RGWPSListNotifsOp {
   }
 
 public:
-  void execute() override;
-  void send_response() override {
+  void execute(const jaeger_tracing::Span* const parent_span = nullptr) override;
```

Comment:
if I understand correctly, you are using parent_span as const, then how would it be adapted to have another reference later or do you intend to keep them null?
```suggestion
  void execute(const jaeger_tracing::Span* parent_span = nullptr) override;
```

Reply:
@ideepika 
its working. what I believe is the new span takes the id of parent_span stores in inside it rather updating any data field of parent_span

Reply:
this means that this parent_span  is optional parameter if I supply none then it will be null but if I supply some value then it will be const variable with the supplied paramtere value

Reply:
@ideepika  
fucn a(parent_span){
Span childspan=childspann(parent_span);
funcb(childspan); now this child span is carrying the load of parent_span for new function,not the parameter parent_span
}

Reply:
okay, so it's not actually the parent that is being passed in these function, in that case can you change parent_span to just span, it might confuse in thinking the current span is the parent span

Reply:
parent_span is still good because they will act as a parent_span parent for the current function in which they have received as parameter